### PR TITLE
Require first name in API/Query tool; Don't require middle name

### PR DESCRIPTION
**Pull request by heymatthenry**
Addresses #528 

* Add validations for first name in Match & Orchestrator APIs
* Remove validation for middle name in Query Tool

### CODE REVIEW

File: match/src/Piipan.Match.Orchestrator/MatchRequest.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ public class MatchQuery
         [JsonProperty("last", Required = Required.Always)]
         public string Last { get; set; }
 
-        [JsonProperty("first", NullValueHandling = NullValueHandling.Ignore)]
+        [JsonProperty("first", Required = Required.Always)]
         [JsonConverter(typeof(NullConverter))]
```
Comment (kingcomma) [inline match/src/Piipan.Match.Orchestrator/MatchRequest.cs:23]: *Nitpick:* Should be able to drop this `NullConverter` now that the property is required in all incoming requests.

File: match/src/Piipan.Match.State/MatchRequest.cs [hunk 2]
```diff
@@ -19,7 +19,7 @@ public class MatchQuery
         [JsonProperty("last", Required = Required.Always)]
         public string Last { get; set; }
 
-        [JsonProperty("first", NullValueHandling = NullValueHandling.Ignore)]
+        [JsonProperty("first", Required = Required.Always)]
         [JsonConverter(typeof(NullConverter))]
```
Comment (kingcomma) [inline match/src/Piipan.Match.State/MatchRequest.cs:23]: *Nitpick:* Should be able to drop this converter as well.

**Review by kingcomma — COMMENTED** [review]
Aside from my nitpicks, I think the SQL that retrieves participant data (see [`Piipan.Match.State.Api.Prepare`](https://github.com/18F/piipan/blob/mh-update-validations/match/src/Piipan.Match.State/Api.cs#L91)) needs a couple adjustments:
- Add an explicit equality condition for first name (eg., `first=@first`) and remove the conditional handling for the now non-existent case when it is `null`
- Consider if the `middle` field should be part of the query at all

I'll let @cjhskippy weigh in on whether or not that second item is out of scope for this story. But my understanding of the desired functionality is that middle names should be accepted but not factored into match determination. E.g.:
- "Foo Bar Doe" should match all of: "Foo Bar Doe", "Foo Doe", and "Foo Baz Doe"
- "Foo Bar" should match all of: "Foo Bar", "Foo Baz Bar", and "Foo * Bar"

As it currently stands, `middle` is an optional field but the exact match algorithm still matches against it (checking equality with `null` in cases when it is not included).

File: match/src/Piipan.Match.Orchestrator/MatchRequest.cs (hunk 1 above)
Comment (heymatthenry) [reply match/src/Piipan.Match.Orchestrator/MatchRequest.cs]: Good nitpick! Don't think I even noticed the `NullConverter`s

File: match/src/Piipan.Match.State/MatchRequest.cs (hunk 2 above)
Comment (heymatthenry) [reply match/src/Piipan.Match.State/MatchRequest.cs]: ✅

**Comment by heymatthenry** [comment]
@kingcomma Per @cjhskippy's comment in Slack, I just removed the null case for `first` and will handle the middle name in another ticket.

**Review by kingcomma — APPROVED** [review]
👍 Looks good to me!

**Pull request merged**